src/components/organisms/LocalReplace.tsx

```diff
@@ -0,0 +1,448 @@
+      }
+
+      const foundMarkText = codeMirror.getRange(
+        markerRange.from,
+        markerRange.to
+      )
+      if (foundMarkText.length >= 1 && preservingCaseReplace) {
+        // if first character was upper case - set the replacement to upper case too
+        codeMirror.replaceRange(
+          foundMarkText.charAt(0) + replaceQuery.substr(1),
+          markerRange.from,
+          markerRange.to,
+          '@ignore'
+        )
+      } else {
+        codeMirror.replaceRange(
+          replaceQuery,
+          markerRange.from,
+          markerRange.to,
+          '@ignore'
+        )
+      }
+
+      return true
+    },
+    [codeMirror, preservingCaseReplace, replaceQuery]
+  )
+
+  const findSelectedMarker = useCallback(() => {
+    for (const marker of codeMirror.getAllMarks()) {
+      if (marker.className === 'marked selected') {
+        return marker
+      }
+    }
+    return null
+  }, [codeMirror])
+
+  const onReplaceCurrentItem = useCallback(() => {
+    const marker = findSelectedMarker()
+    if (!marker) {
+      return
+    }
+
+    const replaceSuccessful = replaceSingleMarker(marker)
+    if (onReplacementFinished != null && replaceSuccessful) {
+      onReplacementFinished()
+    }
+    focusReplaceTextAreaInput(replaceQuery.length)
+  }, [
+    findSelectedMarker,
+    focusReplaceTextAreaInput,
+    onReplacementFinished,
+    replaceQuery.length,
+    replaceSingleMarker,
+  ])
+
+  const onReplaceAll = useCallback(() => {
+    const markers: TextMarker[] = codeMirror.getAllMarks()
+    if (markers.length <= 0) {
+      return
+    }
+
+    let anyReplacementSuccessful = false
+    for (const marker of markers) {
+      const replaceSuccessful = replaceSingleMarker(marker)
+      if (replaceSuccessful) {
+        anyReplacementSuccessful = true
+      }
+    }
+
+    if (onReplacementFinished != null && anyReplacementSuccessful) {
+      onReplacementFinished()
+    }
+  }, [codeMirror, onReplacementFinished, replaceSingleMarker])
+
+  const handleExcludeCurrentItem = useCallback(() => {
+    navigateToNext('next')
+    focusReplaceTextAreaInput(replaceQuery.length)
+  }, [focusReplaceTextAreaInput, navigateToNext, replaceQuery.length])
+
+  const toggleCaseSensitiveReplace = useCallback(() => {
+    onUpdateSearchOptions({ preservingCaseReplace: !preservingCaseReplace })
+    setPreservingCaseReplace(!preservingCaseReplace)
+    focusReplaceTextAreaInput()
+  }, [focusReplaceTextAreaInput, onUpdateSearchOptions, preservingCaseReplace])
+
+  const handleReplaceInputKeyDown = useCallback(
+    (event: KeyboardEvent<HTMLTextAreaElement>) => {
+      switch (event.key) {
+        case 'Enter':
+          event.preventDefault()
+          event.stopPropagation()
+          if (event.ctrlKey && event.shiftKey) {
+            addNewlineToReplaceValue()
+          } else if (!event.repeat) {
+            // todo: [komediruzecki-13/02/2021] If we use repeat debounce should be used
```

Comment:
Can you review this, and let me know if we can remove this note/todo

Reply:
Tested in latest code, it actually works without this condition.
Refactored it!